packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -212,6 +219,47 @@ export const swap = (
   return defaultAcceptanceMsg;
 };
 
+/**
+ * @type Swap
+ * Swap such that both seats gain what they want and lose everything
+ * that they gave. Only good for exact and entire swaps where each
+ * seat wants everything that the other seat has. The benefit of using
+ * this method is that the keywords of each seat do not matter.
+ */
+export const swapExact = (
```

Comment:
Based on this description, perhaps something that is more about that?  `swapAllocations`?

Reply:
or `swapAll`?

Reply:
I think `swapExact` is more precise, because this does not allow giving more than what is wanted by the other side.